xboxlib.c

```diff
@@ -204,6 +205,31 @@ void gigimport(giant g, unsigned char *buff,int len){
 	
 
 }
+#endif
+
+void gigimport(giant g, unsigned char *buff, int len) {
+
+	// copy buffered 'number' into giant's number buffer
+	//int count;
+	memcpy(g->n, buff, len);
+
+	// the giants number buffer stores large numbers as
+	// multiple 16bit numbers, so if the length of our buffer is odd
+	// our number wont fit properly.
+	// to fix this we add a trailing 0x00 byte
+	if (len & 1)
+	{
+		*((unsigned char*)(g->n) + len) = 0x00;
+		len++;
+	}
+	g->sign = len / 2;
+
+	if (g->sign == 0)
```

Comment:
Ideally this case would have to be handled as an early-out in my opinion.

If it does not happen in the code, it should probably just be something like `assert(g->sign > 0); // Value has not been set to 0 if the length was 0`

Reply:
g->sign can be zero only if len is 0 or +/-1 (since +/-0.5 will be truncated to zero), which never happens in practice as explained above, so I would write ```assert(g->sign != 0);```